rails/app/controllers/chapters_controller.rb

```diff
@@ -456,6 +456,9 @@ def show
                                                      :id => "javascript_#{@chapter.id}_#{count}_div")
 
                 chap = order[0];
+                if (order == "3.4.2")
```

Comment:
Let's keep the chapter for snippets as the book chapter. Instead let's use the CHAP option in snippets. For example:
```
Henzs-MacBook-Pro:source-academy henz$ cd sicp
Henzs-MacBook-Pro:sicp henz$ cd xml/
Henzs-MacBook-Pro:xml henz$ grep -r "CHAP=" .
./chapter2/section4/subsection3.xml:      <SNIPPET  CHAP="3">
./chapter2/section4/subsection3.xml:      <SNIPPET CHAP="3">
./chapter2/section4/subsection3.xml:      <SNIPPET CHAP="4">
./chapter2/section4/subsection3.xml:      <SNIPPET CHAP="4">
./chapter2/section4/subsection3.xml:	    <SNIPPET CHAP="4">
./chapter2/section4/subsection3.xml:	    <SNIPPET CHAP="4">
./chapter2/section4/subsection3.xml:      <SNIPPET CHAP="3">
./chapter2/section4/subsection3.xml:	<SNIPPET CHAP="3">
```
This means that all snippets in 3.4 will get "CHAP="3_Concurrent".

Reply:
Didn't know this was possible!